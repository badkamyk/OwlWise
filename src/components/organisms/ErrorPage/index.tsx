import ContentHeading from "../../atoms/MainContentHeading";
import MainContentParagraph from "../../atoms/MainContentParagraph";
import HeadingButton from "../../atoms/HeadingButton";

export default function ErrorPage() {
	return (
		<section className="bg-gray-900 min-h-screen">
			<div className="py-8 px-4 mx-auto max-w-screen-xl lg:py-16 lg:px-6">
				<div className="mx-auto max-w-screen-sm text-center">
					<ContentHeading
						className={"mb-4 text-7xl tracking-tight font-extrabold lg:text-9xl text-white"}
						text={"404"} />
					<MainContentParagraph
						className={"mb-4 text-3xl tracking-tight font-bold text-gray-900 md:text-4xl dark:text-white"}
						text={"Something's missing."} />
					<MainContentParagraph className={"mb-4 text-lg font-light text-gray-500 dark:text-gray-400"}
																text={"Sorry, we can't find that page. You'll find lots to explore on the home page."} />
					<HeadingButton text={"Back to Homepage"} href={"/"} />
				</div>
			</div>
		</section>
	);
}