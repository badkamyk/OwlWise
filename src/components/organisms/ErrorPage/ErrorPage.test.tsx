import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import ErrorPage from "./index";

describe("ErrorPage", () => {
	it("renders ErrorPage correctly with MainHeading, MainContentParagraph and HeadingButton", () => {
		const { container } = render(<ErrorPage />);
		expect(
			container.getElementsByClassName(
				"flex min-h-screen items-center bg-gray-900"
			)
		).toBeTruthy();
		expect(screen.getByTestId("heading-main")).toBeTruthy();
		expect(screen.getAllByTestId("main-content-paragraph")).toHaveLength(2);
		expect(screen.getByTestId("heading-button")).toBeTruthy();
	});
});
