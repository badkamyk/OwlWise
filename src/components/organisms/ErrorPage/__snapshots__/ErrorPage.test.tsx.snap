// Vitest Snapshot v1

exports[`ErrorPage > renders ErrorPage correctly with MainHeading, MainContentParagraph and HeadingButton 1`] = `
<div>
  <section
    class="flex min-h-screen items-center bg-gray-900"
  >
    <div
      class="mx-auto max-w-screen-xl py-8 px-4 lg:py-16 lg:px-6"
    >
      <div
        class="mx-auto max-w-screen-sm text-center"
      >
        <h2
          class="mb-3 text-3xl font-extrabold text-gray-900 dark:text-white md:text-5xl lg:text-6xl"
          data-testid="heading-main"
        >
           
          404
        </h2>
        <p
          class="mb-4 text-white text-xl"
          data-testid="main-content-paragraph"
        >
          Something's missing.
        </p>
        <p
          class="mb-4 text-white text-lg"
          data-testid="main-content-paragraph"
        >
          Sorry, we can't find that page. You'll find lots to explore on the home page.
        </p>
        <button
          class="group relative mb-2 mr-2 inline-flex items-center justify-center overflow-hidden rounded-lg bg-gradient-to-br from-green-400 to-blue-600 p-0.5 text-sm font-medium text-gray-900 hover:text-white focus:outline-none focus:ring-4 focus:ring-green-200 group-hover:from-green-400 group-hover:to-blue-600 dark:text-white dark:focus:ring-green-800"
          data-testid="heading-button"
        >
          <a
            class="relative rounded-md bg-white px-5 py-2.5 transition-all duration-75 ease-in group-hover:bg-opacity-0 dark:bg-gray-900"
            href="/"
          >
            Back to Homepage
          </a>
        </button>
      </div>
    </div>
  </section>
</div>
`;
